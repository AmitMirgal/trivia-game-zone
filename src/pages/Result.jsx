import React, { useMemo } from "react";
import { useLocation, useHistory } from "react-router-dom";
import styled from "styled-components";
import filter from "lodash/filter";
import Card from "../layouts/card/Card";
import CardContent from "../layouts/card/CardContent";
import CardController from "../layouts/card/CardController";
import PrimaryButton from "../components/PrimaryButton";
import { AiOutlineMinus, AiOutlinePlus } from "react-icons/ai";

const ResultPanel = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 16px;
  margin: 16px;
`;

const ResultTitle = styled.h1`
  font-size: 60px;
  font-weight: 700;
  color: #2d3748;
`;

const ResultDescription = styled.div`
  display: grid;
  grid-template-columns: 10% auto;
  align-items: center;
  padding: 4px 0;
  margin: 4px 0;
`;

const Result = () => {
  const location = useLocation();
  const history = useHistory();

  const score = useMemo(() => {
    const count = filter(location.state.result, ["answer", true]);
    return count.length;
  }, [location]);

  const navigateTo = () => {
    history.push("/");
  };

  return (
    <ResultPanel>
      <ResultTitle>{`You scored ${score}/10`}</ResultTitle>

      <Card>
        <CardContent>
          {location.state.result &&
            location.state.result.map((content, index) => (
              <ResultDescription key={`result-screen-content-${index}`}>
                {content.answer ? (
                  <AiOutlinePlus style={{ margin: "4px" }} />
                ) : (
                  <AiOutlineMinus style={{ margin: "4px" }} />
                )}
                <div
                  style={{ color: content.answer ? "green" : "red" }}
                  dangerouslySetInnerHTML={{ __html: content.question }}
                />
              </ResultDescription>
            ))}
        </CardContent>
        <CardController>
          <PrimaryButton label={"Play Again"} primary onClick={navigateTo} />
        </CardController>
      </Card>
    </ResultPanel>
  );
};

export default Result;
