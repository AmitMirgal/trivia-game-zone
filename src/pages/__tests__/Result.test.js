import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Result from "../Result";

jest.mock("react-router-dom", () => ({
  useLocation: jest.fn().mockReturnValue({
    pathname: "/another-route",
    search: "",
    hash: "",
    state: { result: [] },
    key: "result",
  }),
  useHistory: jest.fn().mockReturnValue({}),
}));

describe("<Result/>", () => {
  let wrapper;

  const prepare = () => {
    wrapper = shallow(<Result />);
  };

  it("renders Result component", () => {
    // when
    prepare();

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
