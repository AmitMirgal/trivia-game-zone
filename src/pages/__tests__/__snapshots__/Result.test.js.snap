// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result/> renders Result component 1`] = `
<styled.div>
  <styled.h1>
    You scored 0/10
  </styled.h1>
  <Card>
    <CardContent />
    <CardController>
      <PrimaryButton
        label="Play Again"
        onClick={[Function]}
        primary={true}
      />
    </CardController>
  </Card>
</styled.div>
`;
